src/main/app/controller/courts/CourtFacilitiesController.ts

```diff
@@ -0,0 +1,125 @@
+
+  public async get(
+    req: AuthedRequest,
+    res: Response,
+    updated = false,
+    errorMsg: string[] = [],
+    courtFacilities: Facility[] = null,
+    requiresValidation = true): Promise<void> {
+    if (!courtFacilities) {
+      const slug: string = req.params.slug as string;
+      await req.scope.cradle.api.getCourtFacilities(slug)
+        .then((value: Facility[]) => courtFacilities = value)
+        .catch(() => errorMsg.push(this.getCourtFacilitiesErrorMsg));
+    }
+
+    let allFacilitiesTypes: FacilityType[] = [];
+
+    await req.scope.cradle.api.getAllFacilityTypes()
+      .then((value: FacilityType[]) => allFacilitiesTypes = value)
+      .catch(() => errorMsg.push(this.getFacilityTypesErrorMsg));
+
+    if (!courtFacilities?.some(ot => ot.isNew === true)) {
+      this.addEmptyFormsForNewEntries(courtFacilities);
+    }
+
+    const errors: Error[] = [];
+    for (const msg of errorMsg) {
+      errors.push({text: msg});
+    }
+
+    const pageData: FacilityPageData = {
+      errors: errors,
+      updated: updated,
+      facilitiesTypes: CourtFacilitiesController.getFacilityTypesForSelect(allFacilitiesTypes),
+      courtFacilities: courtFacilities,
+      requiresValidation: requiresValidation
+    };
+
+    res.render('courts/tabs/facilitiesContent', pageData);
+  }
+
+  public async put(req: AuthedRequest, res: Response): Promise<void> {
+    let courtFacilities = req.body.courtFacilities as Facility[] ?? [];
+    courtFacilities.forEach(ot => {
+      ot.isNew = (ot.isNew === true) || ((ot.isNew as any) === 'true');
+      // Workaround for the issue where the the empty row is removed after the previous name selection.
+      if (!ot.name) {
+        ot.name = '';
+      }
+    });
+
+    if(!CSRF.verify(req.body._csrf)) {
+      return this.get(req, res, false, [this.updateErrorMsg], courtFacilities);
+    }
+
+    // Remove fully empty entries
+    courtFacilities = courtFacilities.filter(cf => !this.facilityEntryIsEmpty(cf));
+    const errorMsg: string[] = [];
+
+    if (courtFacilities.some(cf => cf.name === '' || cf.description === '')) {
+      errorMsg.push(this.emptyNameOrDescriptionErrorMsg);
+    }
+
+    if (!validateDuplication(courtFacilities, this.facilityDuplicated)) {
+      errorMsg.push(this.facilityDuplicatedErrorMsg);
+    }
+
+    if (errorMsg.length > 0) {
+      return this.get(req, res, false, errorMsg, courtFacilities);
+    }
+
+    await req.scope.cradle.api.updateCourtFacilities(req.params.slug, courtFacilities)
+      .then((value: Facility[]) => this.get(req, res, true, [], value))
+      .catch(() => this.get(req, res, false, [this.updateErrorMsg], courtFacilities));
+  }
+
+  public async addRow(req: AuthedRequest, res: Response): Promise<void> {
+    const courtFacilities = req.body.courtFacilities as Facility[] ?? [];
+    courtFacilities.forEach(ot => ot.isNew = (ot.isNew === true) || ((ot.isNew as any) === 'true'));
+    this.addEmptyFormsForNewEntries(courtFacilities);
```

Comment:
same here is it possible to give ot.isNew a type such as String and not any?

Reply:
Same as above